File: foundation-domain/ai-platform/Backend/Java/ai-monitoring-service/src/main/java/com/gogidix/aiservices/aimonitoringservice/shared/exception/ValidationException.java
package com.gogidix.aiservices.aimonitoringservice.shared.exception;

/**
 * Exception thrown when validation fails.
 */
public class ValidationException extends BaseDomainException {

    public ValidationException(String message) {
        super(message);
    }

    public ValidationException(String field, String message) {
        super(message);
    }

    @Override
    protected String deriveErrorCode() {
        return "VALIDATION_ERROR";
    }
}

File: management-domain/Sales-department/Backend/Nodes/sales-automation-service/commission-calculator-service/src/shared/exceptions/validation.exception.ts
import { HttpStatus } from '@nestjs/common';
import { BaseException } from './base.exception';

/**
 * Validation exception
 * Thrown when input validation fails
 */
export class ValidationException extends BaseException {
  readonly errorCode = 'VALIDATION_ERROR';

  constructor(message: string, public errors?: Array<{ field: string; message: string }>) {
    super(message, HttpStatus.BAD_REQUEST, { errors });
  }
}
